src/reducers/authenticate.ts

```diff
@@ -1,6 +1,17 @@
+import { types } from "../actions/authenticate";
+
 const initialState = {
-  isAuthenticated: false,
+  isAuthenticated: false
 };
 
-export default function authenticate(state, action) {
-};
+export default function authenticate(state = initialState, action) {
+  switch (action.type) {
+    case types.AUTHENTICATE:
+      return {
+        ...state,
+        isAuthenticated: action.payload
+      };
```

Comment:
As I suggested above, we can divide it to 2 actions LOGIN and LOGOUT:

```js
switch (action.type) {
    case types.LOGIN:
      return { ...state, isAuthenticated: true };
    case types.LOGOUT:
      return { ...state, isAuthenticated: false };
    ...
}
```

Reply:
You can apply boilerplate as well

Reply:
Updated